### ISSUE TITLE
False-positive unsupported-binary-operation on Django Q objects

### ISSUE THREAD

**Issue by masiak2** [PyCQA/pylint#7953]
### Bug description

```python
from django.db import models
from django.db.models import Q


class Author(models.Model):
    name = models.CharField(max_length=200)
    email = models.EmailField()


# this is fine
qs = Author.objects.filter(Q(name="x") | Q(name="y"))

# this throws an unsupported-binary-operation error
qs2 = Author.objects.filter(Q(name="x") | Q(name="y") | Q(name="z"))
```

### Command used

```shell
pylint test_pylint.py
```

### Pylint output

```shell
test_pylint.py:15:28: E1131: unsupported operand type(s) for | (unsupported-binary-operation)
```

### Expected behavior

no error on this line

### Pylint version

```shell
pylint 2.15.8
astroid 2.12.13
Python 3.8.9 (default, Apr 13 2022, 08:48:06) 
[Clang 13.1.6 (clang-1316.0.21.2.5)]
```

### Additional dependencies

```
asgiref==3.5.2
astroid==2.12.13
dill==0.3.6
Django==3.2.16
isort==5.11.2
lazy-object-proxy==1.8.0
mccabe==0.7.0
platformdirs==2.6.0
pylint==2.15.8
pylint-django==2.5.3
pylint-plugin-utils==0.7
pytz==2022.6
sqlparse==0.4.3
tomli==2.0.1
tomlkit==0.11.6
typing_extensions==4.4.0
wrapt==1.14.1
```
also I have checked it on `django==4.1.4` - still the same issue

**Comment by glennmatthews** [comment]
Possible duplicate of #7381?

**Comment by Pierre-Sassoulas** [comment]
Nice catch @glennmatthews, let's close as duplicate I added a comment in the other issue
